_rules/SC4-1-2-iframe-unique-name.md

```diff
@@ -0,0 +1,238 @@
+</iframe>
+<div id="desc-for-title1">List of Reviewers</div>
+<iframe aria-labelledby="desc-for-title1" src="../test-assets/SC4-1-2-iframe-unique-name-doc2.html">
+</iframe>
+```
+
+#### Passed example 5
+
+`iframe` having the same `title` within a given document tree, but one of them is not included in the accessibility tree.
+
+```html
+<iframe aria-hidden='true' title="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+<iframe title="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc2.html">
+</iframe>
+```
+
+#### Passed example 6
+
+`iframe` are allowed to have the same `title` across different document trees. In this example `iframe` with `id` `level2-frame1` has a parent document tree of `iframe` with `id` `level1-frame2`, and does not share the document tree of `iframe` with `id` `level1-frame1`.
+
+```html
+<iframe id="level1-frame1" title="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+<iframe id="level1-frame2" title="List of Contributors 2" src="../test-assets/SC5-1-2-iframe-unique-name-nested.html">
+  <!--
+  Content of document includes an iframe:
+
+  <iframe 
+    id="level2-frame1" title="List of Contributors"     src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+  </iframe>
+  -->
+</iframe>
+```
+
+#### Passed example 7
+
+Usage of `title` attribute to describe the `iframe` content, and there are multiple `iframe` within document tree, with the same `src` and the same accessible name.
+
+```html
+<iframe title="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+<iframe title="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+```
+
+### Failed
+
+#### Failed example 1
+
+Multiple `iframe` within a document tree, with the same accessible name and different src.
+
+```html
+<iframe title="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+<iframe title="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc2.html">
+</iframe>
+```
+
+#### Failed example 2
+
+Multiple frames having the same `aria-label` and different `src` and the same `iframe` title within a given document tree.
+
+```html
+<iframe aria-label="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+<iframe aria-label="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc2.html">
+</iframe>
+```
+
+#### Failed example 3
+
+Multiple frames having the same accessible name given by `title` and `aria-label` with different `src` within a given document tree.
+
+```html
+<iframe title="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+<iframe aria-label="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc2.html">
+</iframe>
+```
+
```

Comment:
All these failed examples now also need ways they fail the expectation or comments that explain why they do so.